### PR TITLE
Add NUMA functional test cases

### PR DESCRIPTION
**Pull request by waynesun09**

### CODE REVIEW

**Comment by waynesun09** [comment]
This depends on:
https://github.com/autotest/virt-test/pull/1999

**Comment by waynesun09** [comment]
new commit 032e375 depends on 
https://github.com/autotest/virt-test/pull/2025

File: libvirt/tests/src/numa/numa_capabilities.py [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+from virttest import utils_libvirtd
+from virttest import utils_misc
+from autotest.client.shared import error
+
+
+def run(test, params, env):
+    """
+    Test capabilities with host numa node topology
+    """
+    status_error = "yes" == params.get("status_error", "no")
```
Comment (Hao-Liu) [inline libvirt/tests/src/numa/numa_capabilities.py]: This variable is not used.

File: libvirt/tests/src/numa/numa_capabilities.py [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    libvirtd = utils_libvirtd.Libvirtd()
+    libvirtd.start()
+    try:
+        new_cap = libvirt_xml.CapabilityXML()
+        if not libvirtd.is_running():
+            raise error.TestFail("Libvirtd is not running")
+        topo = new_cap.cells_topology
+        logging.debug("topo xml is %s", topo.xmltreefile)
+        cell_list = topo.get_cell()
+        numa_info = utils_misc.NumaInfo()
```
Comment (Hao-Liu) [inline libvirt/tests/src/numa/numa_capabilities.py]: I got a warning message here on a non-numa machine:
`Can not find the cpu list information from both numactl and sysfs. Please check your system.`
It seems it is going to check numa node 1 when I only have a node 0 exists.

File: libvirt/tests/src/numa/numa_preferred_undefine.py [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+
+def run(test, params, env):
+    """
+    Test undefine after set preferred numa tuning
+    """
+    bug_url = params.get("bug_url", "")
+    status_error = "yes" == params.get("status_error", "no")
+    vm_name = params.get("vms")
+    vm = env.get_vm(vm_name)
```
Comment (Hao-Liu) [inline libvirt/tests/src/numa/numa_preferred_undefine.py]: `status_error` and `vm` are not used here.

File: libvirt/tests/src/numa/numa_preferred_undefine.py [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+from autotest.client.shared import error
+
+
+def run(test, params, env):
+    """
+    Test undefine after set preferred numa tuning
+    """
+    bug_url = params.get("bug_url", "")
+    status_error = "yes" == params.get("status_error", "no")
+    vm_name = params.get("vms")
```
Comment (Hao-Liu) [inline libvirt/tests/src/numa/numa_preferred_undefine.py]: There could be multiple entries in `vms`. I think `main_vm` suits your case more.

File: libvirt/tests/src/numa/numad_vcpupin.py [hunk 5]
```diff
@@ -0,0 +1,68 @@
...
+
+
+def run(test, params, env):
+    """
+    Test vcpupin while numad is running
+    """
+    vcpu_placement = params.get("vcpu_placement")
+    bug_url = params.get("bug_url", "")
+    status_error = "yes" == params.get("status_error", "no")
+    vm_name = params.get("vms")
```
Comment (Hao-Liu) [inline libvirt/tests/src/numa/numad_vcpupin.py]: Here too.

**Comment by Hao-Liu** [comment]
Except some nitpicks, this PR LGTM.

File: libvirt/tests/src/numa/numa_capabilities.py (hunk 2 above)
Comment (waynesun09) [inline libvirt/tests/src/numa/numa_capabilities.py]: @Hao-Liu it works on single node system, the warning is ok.

**Comment by waynesun09** [comment]
@Hao-Liu updated according to your comments. Thanks.

**Comment by Hao-Liu** [comment]
LGTM. Merging. Thanks @waynesun09.

**Pull request merged**